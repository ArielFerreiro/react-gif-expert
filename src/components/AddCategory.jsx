import PropTypes from 'prop-types';
import { useState } from "react";

export const AddCategory = ({ onAddCategory }) => {

    const [ inputValue, setInputValue ] = useState('');

    const onInputChange = (e) => {
        setInputValue( val => e.target.value);
    }

    const onSubmit = ( e ) => {
        e.preventDefault();

        if (inputValue.trim().length <= 1) return;
        onAddCategory(inputValue.trim());
        setInputValue( val => '');
    }

    return (
        <form aria-label="form" onSubmit={ (event) => onSubmit(event)}>
            <input 
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={ (event) => onInputChange(event) }
            />
            <button>Agregar</button>
        </form>
    );
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}