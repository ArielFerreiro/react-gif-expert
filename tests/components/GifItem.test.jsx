import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Test in Component <GifItem />', () => {

    const title = 'Test';
    const url = 'https://www.google.com';

    test('Should match Snapshot', () => {

        const { container } = render( <GifItem title={title} url={url} />);
        expect( container ).toMatchSnapshot();
        //screen.debug();
    });

    test('Most show the image of the url and the indicated ALT', () => {

        render( <GifItem title={title} url={url} />);
        
        const { src, alt} = screen.getByRole('img');

        expect( src ).toContain(url);
        expect( alt ).toBe(title);

    });

    test('The title must be in a paragraph', () => {

        render( <GifItem title={title} url={url} />);
        expect( screen.getByText( title ) ).toBeTruthy();

    });


});