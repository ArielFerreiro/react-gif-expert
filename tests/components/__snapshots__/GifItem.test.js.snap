// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Component <GifItem /> Should match Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test"
      src="https://www.google.com"
    />
    <p>
       
      Test
       
    </p>
  </div>
</div>
`;
